### ISSUE TITLE
[BGP-Unnumbered] Only one of parallel BGP-Unnumbered sessions is up

### ISSUE THREAD

**Issue by Hedgehog-Guru** [sonic-net/sonic-buildimage#12042]
#### Description

Two switches connected using several L3 ports. No IPv4 or IPv6 addresses. Just IPv6 link-locals. Due all ports have the same MAC, all IPv6-LL are the same on the switch,
Each port has BGP-Unnumbered peering configured. Only one BGP session is up, while all are expected.

#### Steps to reproduce the issue:
1. Enable ports. Speed, FEC, etc
2. Enable IPv6-LL  `config ipv6 enable link-local`
3. Add split mode to config_db to avoid FRR config rewrite during reload `"docker_routing_config_mode": "split"`
4. On both switches add BGP unnumbered config for each port to FRR. 
 ```
router bgp 1
 bgp router-id 1.1.1.1
 no bgp ebgp-requires-policy
 neighbor Ethernet48 interface remote-as external
 neighbor Ethernet49 interface remote-as external
 neighbor Ethernet50 interface remote-as external
 neighbor Ethernet51 interface remote-as external
```
5. Check BGP sessions status `show bgp summary`

#### Describe the results you received:
Only one BGP session is in established state

#### Describe the results you expected:
All BGP sessions are up

#### Output of `show version`:

```
SONiC Software Version: SONiC.202205_rc.7-4d3ba951a_Internal
Distribution: Debian 11.4
Kernel: 5.10.0-12-2-amd64
Build commit: 4d3ba951a
Build date: Thu Sep  1 15:13:23 UTC 2022
Built by: sw-r2d2-bot@r-build-sonic-ci02-241

Platform: x86_64-mlnx_msn3700c-r0
HwSKU: ACS-MSN3700C
ASIC: mellanox
ASIC Count: 1
Serial Number: MT1935X01905
Model Number: MSN3700-CS2F
Hardware Revision: A4
Uptime: 14:12:46 up 17 min,  2 users,  load average: 0.90, 0.90, 0.66
Date: Tue 06 Sep 2022 14:12:46
```
#### Output of `show techsupport`:

[sonic_dump_qa-eth-vt01-4-3700c_20220906_141235.tar.gz](https://github.com/sonic-net/sonic-buildimage/files/9538423/sonic_dump_qa-eth-vt01-4-3700c_20220906_141235.tar.gz)

#### Additional information you deem important (e.g. issue happens only occasionally):

If user run tcpdump (promisc on/off) on any port where BGP is down, session will be established.

**Comment by dgsudharsan** [comment]
@zhangyanzhao This is not triaged today. Can we please assign it

@adyeung  I believe broadcom is the feature owner for unnumbered and link local. Can you please assign someone to investigate?

**Comment by santoshdoke** [comment]
Hello, 

Based on the tech support, it looks like the neighbour resolution is completed for one peer only and for that peer BGP adjacency is established. Ethernet48 is the link where neighbour is resolved and BGP is established. For other 3 interfaces, neighbour is in stale state, and BGP is not receiving/transmitting any packets. Please check ping to peer's IPv6 link-local addresses using interface option, once that works, neighbor should be resolved and BGP adjacency should work on all interfaces.

fe80::7efe:90ff:fe40:9e00 dev Ethernet51 lladdr 7c:fe:90:40:9e:00 router STALE
fe80::7efe:90ff:fe40:9e00 dev Ethernet49 lladdr 7c:fe:90:40:9e:00 router STALE
fe80::7efe:90ff:fe40:9e00 dev Ethernet50 lladdr 7c:fe:90:40:9e:00 router STALE
fe80::7efe:90ff:fe40:9e00 dev Ethernet48 lladdr 7c:fe:90:40:9e:00 router REACHABLE

**Comment by Hedgehog-Guru** [comment]
Here are CLI output and sysdump shows that neighbors could be resolved, but BGP sessions are still in down state

[CLI-log1.txt](https://github.com/sonic-net/sonic-buildimage/files/9633254/CLI-log1.txt)
[sonic_dump_qa-eth-vt01-4-3700c_20220923_144012.tar.gz](https://github.com/sonic-net/sonic-buildimage/files/9633257/sonic_dump_qa-eth-vt01-4-3700c_20220923_144012.tar.gz)

**Comment by santoshdoke** [comment]
Hello,

If the IPv6 link-local neighbors are resolved, then there is no issue with Link-local addressing and reachability.  I suspect the issue is with BGP FRR code (may be missing some fixes on these lines). BGP should detect the new neighbor and initiate the TCP connection, but that doesn't seem to be happening. Can you please involve BGP expert to this thread.

**Comment by adyeung** [comment]
@dgsudharsan @zhangyanzhao can someone take a look from BGP to see why the tcp session is not initiated?

**Comment by madhupalu** [comment]
@dgsudharsan how does this issue addressed ?  are there FRR update or any other fix went in to address this issue? 
Can you share insights how does it fixed ?
Thanks

**Comment by dgsudharsan** [comment]
@madhupalu This was fixed by FRR 8.5.1 upgrade. No other additional fix is required. FRR 8.5+ introduced support to have multiple unnumbered sessions between two devices.

**Comment by madhupalu** [comment]
Thank you @dgsudharsan
